### PR TITLE
Make help command auto-updating

### LINKED ISSUE #22: Make the help command auto-updating.
Make the `help` command of the bot better, because currently, I assume whenever a new command is programmed, you manually have to add the name and description to the `help` command embed. Instead you can use the library to get all the command names with their description automatically and then iterate through them, appending fields to the embed. But an embed can only have 25 fields, so after each 25 index, you DM whoever invoked the command with the current batch. After that, the fields can be cleared and new command names/description can be added.

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Help command will now do the following:

- [x] Automatically scan the files in the commands directory for commands.
- [x] Create embed with fields, each containing a command's prefix and description.
- [x] Skip any file in the commands directory that is missing either the prefix or description property. 
- [x] Trim the name and value of each embed field to ensure it does not exceed the limits. 

Closes #22

### CODE REVIEW

**Comment by Twaha-Rahman** [comment]
Looks like an interesting addition. Well done @nhcarrigan !

**Comment by naomi-lgbt** [comment]
Thanks! Not sure how well it will mesh with #32 but we can tweak it as needed I'm sure.

File: src/commands/help.js [hunk 1]
```diff
@@ -1,44 +1,48 @@
...
+      };
 
-          //for spacing, leave this one at bottom.
-          { name: '\u200B', value: '\u200B' }
-        )
-        .setFooter('I am not affiliated with FreeCodeCamp in any way.');
+      fs.readdir(__dirname, (error, files) => {
+        if (error) {
+          console.error(error);
+        }
```
Comment (bradtaniguchi) [inline src/commands/help.js]: Change this to the `promise` version:
https://nodejs.org/dist/latest-v10.x/docs/api/fs.html

File: src/commands/help.js [hunk 2]
```diff
@@ -1,44 +1,48 @@
...
+        for (let i = 0; i < files.length; i++) {
+          let filename = files[i];
+          let lookup = require(`./${filename}`);
+          if (lookup.prefix == undefined || lookup.description == undefined) {
```
Comment (bradtaniguchi) [inline src/commands/help.js]: Don't use `==`, use `===`, or just check for "falsy":
```js
if (!lookup.prefix || !lookup.description) {
```

File: src/commands/help.js [hunk 3]
```diff
@@ -1,44 +1,48 @@
...
-          { name: '\u200B', value: '\u200B' }
-        )
-        .setFooter('I am not affiliated with FreeCodeCamp in any way.');
+      fs.readdir(__dirname, (error, files) => {
+        if (error) {
+          console.error(error);
+        }
+        for (let i = 0; i < files.length; i++) {
+          let filename = files[i];
+          let lookup = require(`./${filename}`);
```
Comment (bradtaniguchi) [inline src/commands/help.js]: Don't use `let` if you can use `const`. (this should be made a linter rule, but that can be done later)

**Review by bradtaniguchi — CHANGES_REQUESTED** [review]

File: src/commands/help.js (hunk 2 above)
Comment (naomi-lgbt) [reply src/commands/help.js]: Done :)

File: src/commands/help.js (hunk 3 above)
Comment (naomi-lgbt) [reply src/commands/help.js]: Wasn't sure if I could use const within the loop. Corrected.

File: src/commands/help.js (hunk 1 above)
Comment (naomi-lgbt) [reply src/commands/help.js]: Believe this is implemented correctly now.

File: src/commands/help.js [hunk 4]
```diff
@@ -1,44 +1,49 @@
...
+        footer: { text: 'I am not affiliated with FreeCodeCamp in any way.' }
+      };
 
-          //for spacing, leave this one at bottom.
-          { name: '\u200B', value: '\u200B' }
-        )
-        .setFooter('I am not affiliated with FreeCodeCamp in any way.');
+      fsPromises
+        .readdir(__dirname)
+        .then((result) => {
```
Comment (bradtaniguchi) [inline src/commands/help.js]: Change to `async/await`, we are trying not to use `.then` and `.catch` when possible.
Also change `result` to `results`, as its an array, and thus should be named plural.

File: src/commands/help.js [hunk 5]
```diff
@@ -1,44 +1,49 @@
...
+            if (!lookup.prefix || !lookup.description) {
+              continue;
+            } else {
+              const fieldObj = {
+                name: lookup.prefix.substring(0, 255),
+                value: lookup.description.substring(0, 1023)
+              };
+              helpEmbed.fields.push(fieldObj);
+            }
+          }
```
Comment (bradtaniguchi) [inline src/commands/help.js]: I'm not a fan of the classic for loop, `.forEach`, or [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) is less to type, and is less likely to get screwed up.

**Review by bradtaniguchi — CHANGES_REQUESTED** [review]

File: src/commands/help.js (hunk 5 above)
Comment (naomi-lgbt) [reply src/commands/help.js]: Changed to .forEach

File: src/commands/help.js (hunk 4 above)
Comment (naomi-lgbt) [reply src/commands/help.js]: Think I got what you were saying.

File: src/commands/help.js [hunk 6]
```diff
@@ -1,46 +1,42 @@
...
+        if (lookup.prefix && lookup.description) {
+          const fieldObj = {
+            name: lookup.prefix.substring(0, 255),
+            value: lookup.description.substring(0, 1023)
+          };
+          helpEmbed.fields.push(fieldObj);
+        }
+      });
+      await message.author.send({ embed: helpEmbed });
+    });
```
Comment (bradtaniguchi) [inline src/commands/help.js]: Update to:
```js
const results = await fsPromises.readdir(__dirname);
console.log(results);
//... rest of code
```

Comment (naomi-lgbt) [reply src/commands/help.js]: Should be good to go now!

**Review by bradtaniguchi — APPROVED** [review]
LGTM 👍 🚀 :shipit:

**Pull request merged**